web/docs/project/custom-vite-config.md

```diff
@@ -111,3 +111,58 @@ If you, for example, want to serve the client from a different path than `/`, yo
     ```
   </TabItem>
 </Tabs>
+
+## Chrome DevTools automatic workspace mapping {#devtools-workspace}
+
+Want edits you make in Chrome DevTools Sources panel to save directly to your project files? You can enable [Chrome DevTools Workspace](https://developer.chrome.com/docs/devtools/workspace/) support in development with the community plugin **vite-plugin-devtools-json** (no Wasp fork required).
+
+1. Install the plugin as a **dev-dependency**:
+
+```bash
+npm add -D vite-plugin-devtools-json
+```
+
+2. Extend your `vite.config.{ts,js}`:
+
+<Tabs groupId="js-ts">
+  <TabItem value="js" label="JavaScript">
+
+```js title="vite.config.js"
+import { devtoolsJson } from 'vite-plugin-devtools-json'
+
+export default {
+  plugins: [
+    react(),              // already present
+    devtoolsJson({
+      // tell Chrome which folder to map – project root works great
+      root: __dirname
+    })
+  ]
+}
+```
+
+  </TabItem>
+  <TabItem value="ts" label="TypeScript">
+
+```ts title="vite.config.ts"
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react-swc'
+import { devtoolsJson } from 'vite-plugin-devtools-json'
+
+export default defineConfig({
+  plugins: [
+    react(),
+    devtoolsJson({ root: __dirname })
+  ]
+})
+```
+  </TabItem>
+</Tabs>
```

Comment:
- We have a plugin that lets us write only the TS version and converts it to JS automatically. You just write a normal code block and add the `auto-js` attribute to it.

- This file is an ES Module, not a CommonJS file, so `__dirname` is not the recommended way to get the dirname. We should use [`import.meta.dirname`](https://nodejs.org/docs/latest-v22.x/api/esm.html#importmetadirname) instead.

- We don't require users to add the `react()` plugin, that should be taken out.

- `devtoolsJson` is a _default_ export

~~~suggestion
```ts title="vite.config.ts" auto-js
import { defineConfig } from 'vite'
import devtoolsJson from 'vite-plugin-devtools-json'

export default defineConfig({
  plugins: [
    devtoolsJson({ root: import.meta.dirname })
  ]
})
```
~~~